packages/hierarchy-browser/src/hooks/useUpdatedHierarchyProvider.ts

```diff
@@ -0,0 +1,60 @@
+/*!
+ * Copyright (c) Microsoft. All rights reserved.
+ * Licensed under the MIT license. See LICENSE file in the project.
+ */
+import { useMemo, useState, useEffect, useCallback } from 'react'
+import { HierarchyDataProvider } from '../common/dataProviders/HierachyDataProvider'
+import {
+	CommunityId,
+	ICommunityDetail,
+	IEntityDetail,
+	IHierarchyNeighborResponse,
+	ILoadEntitiesAsync,
+	ILoadNeighborCommunitiesAsync,
+	ILoadParams,
+	INeighborCommunityDetail,
+} from '../types'
+export function useUpdatedHierarchyProvider(
+	communities: ICommunityDetail[],
+	hierachyDataProvider?: HierarchyDataProvider,
+	entities?: IEntityDetail[] | ILoadEntitiesAsync,
+	neighbors?: INeighborCommunityDetail[] | ILoadNeighborCommunitiesAsync,
+): [
+	boolean,
+	(
+		params: ILoadParams,
+		communityId: CommunityId,
+	) => Promise<IHierarchyNeighborResponse>,
+] {
+	const [isNeighborsLoaded, setIsNeighborsLoaded] = useState<boolean>(false)
+	useEffect(() => {
+		hierachyDataProvider && hierachyDataProvider.updateCommunities(communities)
```

Comment:
The ?. operator can be used here instead of steeltoeing:
`hierarchyDataProvider?.updateCommunities(...)`